zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithSuspensionInstructionTest.java

```diff
@@ -0,0 +1,560 @@
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    // then
+    final var result =
+        RecordingExporter.processInstanceRecords()
+            .onlyEvents()
+            .withProcessInstanceKey(processInstanceKey)
+            .limit(processId, ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .filter(record -> record.getValue().getElementId().equals(processId));
+
+    assertThat(result)
+        .extracting(Record::getIntent, record -> record.getValue().getElementId())
+        .contains(Tuple.tuple(ProcessInstanceIntent.ELEMENT_SUSPENDED, processId));
+
+    final var processInstanceRecord =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    ENGINE.writeRecords(
+        RecordToWrite.command()
+            .key(processInstanceKey)
+            .processInstance(ProcessInstanceIntent.CANCEL, processInstanceRecord.getValue()));
+
+    final var terminatedEvent =
+        RecordingExporter.processInstanceRecords()
+            .onlyEvents()
+            .withProcessInstanceKey(processInstanceKey)
+            .limit(processId, ProcessInstanceIntent.ELEMENT_TERMINATED)
+            .getFirst();
+
+    assertThat(terminatedEvent).isNotNull();
+  }
+
+  @Test
+  public void shouldNotAllowToMigrateSuspendedProcessInstance() {
+    // given
+    final String processId = "process";
+    final String elementToSuspend = "element";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .manualTask(elementToSuspend)
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    final long updatedProcessDefinitionKey =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .manualTask(elementToSuspend)
+                    .manualTask("anotherManualTask")
+                    .endEvent()
+                    .done())
+            .deploy()
+            .getKey();
+
+    // when
+    final var migrationRecord =
+        new ProcessInstanceMigrationRecord()
+            .setProcessInstanceKey(processInstanceKey)
+            .setTargetProcessDefinitionKey(updatedProcessDefinitionKey);
+
+    ENGINE.writeRecords(RecordToWrite.command().key(processInstanceKey).migration(migrationRecord));
```

Comment:
🔧 Same as above, you can simply use the test client to migrate the process instance.